### ISSUE TITLE
ember 2.11 - cannot pass class to store method

### ISSUE THREAD

**Issue by jamesdixon** [lytics/ember-data-model-fragments#229]
Hi,

I tried running Ember Data 2.11 Canary alongside Ember 2.9 and Ember Data Model Fragments 2.3.2 and received the following error:

```
Assertion Failed: Passing classes to store methods has been removed. Please pass a dasherized string instead of scout-app@model:address:
```

I'm still trying to track down where it occurred specifically, but wanted to get this posted. I'll report back with additional details when/if I find them.

Thanks!

**Comment by workmanw** [comment]
@jamesdixon Thanks! Let us know what you find. A few notes / FYIs.  

Officially speaking, this addon doesn't support mixing `ember` and `ember-data` versions. If there is an issue we can easily fix, we'll definitely do so. In the past we've run into a few very complicated issues that stem from using older ember with newer ember-data. 

There is a known issue with Ember Data 2.11 beta that causes a stack exception due to bad recursion. You can read more about that here: #227. It doesn't sound like you're hitting that issue so you might not have to worry about it. But if you do hit that issue, you can easily pull in that patch by updating your package.json to:

```
"ember-data-model-fragments": "lytics/ember-data-model-fragments#pull/227/head"
```

**Comment by workmanw** [comment]
@jamesdixon Did you ever get anything figured out here?

**Comment by jamesdixon** [comment]
@workmanw I haven't had a chance yet as I've been traveling. I'll give it a go again and do some investigation as soon asi have some time. Thanks!

**Comment by workmanw** [comment]
@jamesdixon Sounds good! I'll leave this open for a while. No hurry.

**Comment by jamesdixon** [comment]
@workmanw thanks for your patience. Here's what I discovered:

[This line](https://github.com/lytics/ember-data-model-fragments/blob/master/addon/ext.js#L51) appears to be the offender. In my case, the string "address" is being passed to `createFragment` as `modelName`. The call to `InternalModel` on the line referenced above completes successfully, but when I look at the object, the `modelName` property is a reference to the `address` model/fragment rather than a string, which appears to be what ED is expecting.

[This line](https://github.com/lytics/ember-data-model-fragments/blob/master/addon/ext.js#L61) is actually what triggers the error.

I'm guessing there have been some changes to that private API. Thoughts?

**Comment by jamesdixon** [comment]
@workmanw just an FYI that I updated my Ember application to 2.10, but not moving to 2.11-beta just yet.

**Comment by workmanw** [comment]
@jamesdixon -- Thanks a lot! Let me see what I can dig up. I know some of the ember-data internals have been iterated and tweaked lately. Just to be clear, you're seeing this issue when calling `createFragment` w/ `ember 2.11`, `ember-data 2.11` and `ember-data-model-fragments 2.3.x`?

**Comment by jamesdixon** [comment]
@workmanw I've tried a couple of different configs:

Ember 2.10 w/ ED Canary
Ember 2.11-beta 1 w/ ED 2.11 beta
Ember 2.10 w/ ED 2.11 beta

Needless to say, I've run into an issue with each of those configs.

I'm running ember-data-model-fragments 2.3.3.

Do you have a configuration that is working as expected for you?

**Comment by workmanw** [comment]
@jamesdixon I'm seeing this deprecation now with our test suite against `ember-data-2.12.0-beta.1`. 

However when I use PR #227 with ember/ember-data-2.11 it works as expected. I'll be cutting `ember-data-model-fragments-2.11.0` in the very near future that should work with ember/ember-data-2.11. After that, I'll start investigating this deprecation for our ember-data-beta (2.12) support.

**Comment by jamesdixon** [comment]
@workmanw thanks for the update! I'll give things a try w/ 2.11

**Comment by jamesdixon** [comment]
@workmanw just wanted to give you a quick update: with ED 2.11 being officially released, I gave it a shot using Ember 2.10, Ember Data 2.11 and the #227 as you suggested above and my app is now running flawlessly.

**Comment by workmanw** [comment]
v2.11.0 has been released. I'm going to close this. Thanks a lot of your help.
